src/metadata.rs

```diff
@@ -565,7 +562,10 @@ impl SignatureScheme {
     fn verify(&self, pub_key: &KeyValue, msg: &[u8], sig: &SignatureValue) -> Result<(), Error> {
         match self {
             &SignatureScheme::Ed25519 => {
-                if ed25519::verify(msg, &pub_key.0, &sig.0) {
+                if ring::signature::verify(
+                    &ED25519,
+                    Input::from(&pub_key.0), Input::from(msg), Input::from(&sig.0)
+                ).is_ok() {
                     Ok(())
                 } else {
                     Err(Error::VerificationFailure("Bad signature".into()))
```

Comment:
Usually there is a better way of doing things than using `is_err()`. For example:

`ring::signature::verify(&ED25519, Input::from(&pub_key.0), Input::from(msg), Input::from(&sig.0), Input::from(msg), Input::from(&sig.0)).map_err(|_| Error::VerificationFailure("Bad signature".into()))`.